pydantic/_hypothesis_plugin.py

```diff
@@ -284,7 +285,21 @@ def resolve_confloat(cls):  # type: ignore[no-untyped-def]
         assert max_value is None, 'Set `lt` or `le`, but not both'
         max_value = cls.lt
         exclude_max = True
-    return st.floats(min_value, max_value, exclude_min=exclude_min, exclude_max=exclude_max, allow_nan=False)
+
+    if cls.multiple_of is None:
+        return st.floats(min_value, max_value, exclude_min=exclude_min, exclude_max=exclude_max, allow_nan=False)
+
+    if min_value is not None:
+        min_value = math.ceil(min_value / cls.multiple_of)
+        if exclude_min:
+            min_value = min_value + 1
+    if max_value is not None:
+        assert max_value >= cls.multiple_of, 'Cannot build model with max value smaller than multiple of'
```

Comment:
This assertion restricts the generation more than needed. For example, `max_value=-2.0` and `multiple_of=-1.0` will fail this assertion, but if the plugin generates `-3.0`, it satisfies the constraints.

Generally, I suggest taking a look at `multipleOf` [implementation](https://github.com/Zac-HD/hypothesis-jsonschema/blob/master/src/hypothesis_jsonschema/_canonicalise.py#L699) in `hypothesis-jsonschema` (there are multiple places in the linked file) as it handles all these constraints.

Reply:
thanks, will take a look